action/copy.go

```diff
@@ -193,6 +192,18 @@ func CopyNotSeekableFileToRemote(r io.Reader, objectKey string) (total int64, er
 	return total, nil
 }
 
+// CopySeekableFileToRemote will copy a seekable file to remote.
+func CopySeekableFileToRemote(r *os.File, objectKey string) (total int64, err error) {
```

Comment:
Let's create a new interface called `ReadCloseSeeker`, and use this instead of `os.File`